[vtk] Fix warning in VTK::CommonDataModel

vtkBoundingBox: -Wold-style-cast

This patch is no longer needed upstream, and can be removed
from drake the next time we update VTK.  The VTK module
upstream has been rewritten and the patch will no longer apply.

--- Common/DataModel/vtkBoundingBox.h
+++ Common/DataModel/vtkBoundingBox.h
@@ -86,7 +86,7 @@
   static void ComputeBounds(vtkPoints* pts, const unsigned char* ptUses, double bounds[6]);
   static void ComputeBounds(
     vtkPoints* pts, const std::atomic<unsigned char>* ptUses, double bounds[6]);
-  void ComputeBounds(vtkPoints* pts) { this->ComputeBounds(pts, (unsigned char*)nullptr); }
+  void ComputeBounds(vtkPoints* pts) { this->ComputeBounds(pts, static_cast<unsigned char*>(nullptr)); }
   void ComputeBounds(vtkPoints* pts, unsigned char* ptUses)
   {
     double bds[6];
